### PR TITLE
shopping list with hooks

### PR DESCRIPTION
**Pull request by Kavster92**
[DEMO LINK](https://Kavster92.github.io/react_goods-selector-hooks/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -15,71 +15,80 @@ export const goods = [
...
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+export const App: React.FC = () => {
+  const [selectedGood, setSelectedGood] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  const goodsList = goods.map(good => (
```
Comment (Grayfox147) [inline src/App.tsx:21]: Really interesteting you save the whole markup to a variable.

File: src/App.tsx [hunk 2]
```diff
@@ -15,71 +15,80 @@ export const goods = [
...
+          (!selectedGood
+            ? 'No goods selected'
+            : `${selectedGood} is selected`
+          )
+        }
 
-        <tr data-cy="Good">
-          <td>
+        {
+          selectedGood && (
```
Comment (Grayfox147) [inline src/App.tsx:75]: good implementation!

**Review by Grayfox147 — COMMENTED** [review]
Overall quite interesting approach, about savint the markup and the deploy it at the end, the Demo link is not working I believe you copy a previous one probably! but besides that seems all good :D

**Review by Grayfox147 — CHANGES_REQUESTED** [review]
The demo Link is wrong it sends me to another task, if you could update it will be perfect!

File: src/App.tsx (hunk 1 above)
Comment (naziamoff) [reply src/App.tsx:21]: yeah, the magic of react. Still, you shouldn't do it, it decreases readability

File: src/App.tsx [hunk 3]
```diff
@@ -15,71 +15,80 @@ export const goods = [
...
-      Jam is selected
+  const goodsList = goods.map(good => (
+    <tr
+      data-cy="Good"
+      key={good}
+      className={
+        (selectedGood === good)
+          ? ('has-background-success-light')
+          : ''
+      }
```
Comment (naziamoff) [inline src/App.tsx:29]: [classnames](https://www.npmjs.com/package/classnames)

File: src/App.tsx [hunk 4]
```diff
@@ -15,71 +15,80 @@ export const goods = [
...
-        </tr>
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {
+          (!selectedGood
+            ? 'No goods selected'
+            : `${selectedGood} is selected`
+          )
+        }
```
Comment (naziamoff) [inline src/App.tsx:72]:
```suggestion
        {!selectedGood
          ? 'No goods selected'
          : `${selectedGood} is selected`}
```

File: src/App.tsx (hunk 2 above)
Comment (naziamoff) [inline src/App.tsx:75]:
```suggestion
        {selectedGood && (
```

**Review by naziamoff — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 3 above)
Comment (Kavster92) [reply src/App.tsx:29]: please contact me to explain this one in more detail, it has been installed.

File: src/App.tsx [hunk 5]
```diff
@@ -15,71 +15,78 @@ export const goods = [
...
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+      <table className="table">
+        <tbody>
+          {goods.map(good => (
+    <tr
```
Comment (naziamoff) [inline src/App.tsx:47]: what happened to formatting?

File: src/App.tsx (hunk 3 above)
Comment (naziamoff) [reply src/App.tsx:54]: https://www.npmjs.com/package/classnames#usage

File: src/App.tsx [hunk 6]
```diff
@@ -15,71 +15,78 @@ export const goods = [
...
-    <h1 className="title">No goods selected</h1>
+export const App: React.FC = () => {
+  const [selectedGood, setSelectedGood] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {
```
Comment (naziamoff) [inline src/App.tsx:24]: formatting is not fixed

**Review by naziamoff — CHANGES_REQUESTED** [review]

**Review by Mohameddahi — APPROVED** [review]

**Review by naziamoff — APPROVED** [review]

**Review by naziamoff — APPROVED** [review]

**Pull request closed without merging**